microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManager.java

```diff
@@ -289,24 +288,36 @@ private IJavaProject getJavaProject(IJavaProject javaProject, boolean excludeTes
 		}
 		int length = getPropertiesProviders().size();
 		SubMonitor mainMonitor = monitor;
-		IClasspathEntry[] resolvedClasspath = ((JavaProject) javaProject).getResolvedClasspath();
-		List<IClasspathEntry> newClasspathEntries = new ArrayList<>();
+		BuildingScopeContext context = new BuildingScopeContext(javaProject, excludeTestCode, scopes,
+				ArtifactResolver.DEFAULT_ARTIFACT_RESOLVER);
+		beginBuildingScope(context, mainMonitor);
 		for (int i = 0; i < length; i++) {
 			mainMonitor.subTask("Contributing to classpath for provider (" + (i + 1) + "/" + length + ")");
 			SubMonitor subMonitor = mainMonitor.split(1);
 			IPropertiesProvider provider = getPropertiesProviders().get(i);
-			provider.contributeToClasspath(javaProject, resolvedClasspath, excludeTestCode, scopes,
-					ArtifactResolver.DEFAULT_ARTIFACT_RESOLVER, newClasspathEntries, subMonitor);
+			provider.contributeToClasspath(context, subMonitor);
 			subMonitor.done();

```

Comment:
Could we have this in another method like `beginBuildingScope()` and `endBuildingScope()`, just to clean things up and be consistent?

Like:
```
private void contributeToClasspath(BuildingScopeContext context, IProgressMonitor monitor) throws JavaModelException {
	int length = getPropertiesProviders().size();
	for (int i = 0; i < length; i++) {
		monitor.subTask("Contributing to classpath for provider (" + (i + 1) + "/" + length + ")");
		SubMonitor subMonitor = ((SubMonitor)monitor).split(1);
		IPropertiesProvider provider = getPropertiesProviders().get(i);
		provider.contributeToClasspath(context, subMonitor);
		subMonitor.done();
	}
}
```

Reply:
Why do you want to do that? The main idea (like collection of properties) is to have a begin (o initialize something), process (do some work which can take time) and end (dispose something) methods.

Reply:
@xorye updated